haroonsheikh/HAROON_SHEIKH.md

```diff
@@ -0,0 +1,47 @@
+
+The experience above showed how important it is for a tester to not only understand the requirements of the system-under-test but understanding how the system is doing what it is.  Once acquired some technical knowledge of the system I was testing, was then able to find critical issues as well as gather traces using various tools such as Wireshark to aid in more reproducible bugs.
+
+From first testing role to now a technical tester with variety of experiences, following are few areas I have continuously practiced contributing more productively:
+
+## Knowing the application and test requirements
+
+Whether you're a black-box tester or a white-box tester, it's important to know the requirements of an application to test and it’s even more important to discuss the test requirements.  "What are the test requirements?" has been on top of my "feature/story kick-off" template to discuss during planning how this new feature can be developed with testability in mind.  In my experience, I have found that testable features of an application are easy and less costly to maintain with less regressions.
+
+I have worked on several projects where there are no requirements documented or it's just a "bad requirement" which makes it difficult to test with confidence as there's a bunch of assumptions and no baseline.  When requirements are not clear, it’s good practice to handle these situations with plenty of exploratory testing prioritizing user experience.
+
+## Knowing the technology stack
+
+One area that has helped in getting most amount of test coverage is an understanding of the technology stack without knowing anything about coding.  I have This enables targeted testing of not only front-end components but also backend technologies including databases.
+
+- It really aids in understanding the inputs and outputs of the system and can easily see where in the automation pyramid a particular test needs to reside.  I.e., Do I really need a test at UI (User Interface) layer when it can easily be covered by a more stable integration/unit test.
+- It's important to understand how we can monitor the application behaviour by looking at the logs when an unexpected behaviour occurs at each layer of the stack and helps in tracing the flow or sequence through a system.
+- Just knowing the technology can help with identifying potentially interesting variables to explore with testing.  For example, the version of a particular technology used can help with picking on any present security vulnerabilities.  If you're looking to automate testing, then this can also help in deciding what tools and languages to choose for the tests. E.g., If the front-end application is written is `react-js` then writing the integration tests in `testcafe` maybe the preferred to stay in sync with the programming language and shareability of the selectors, etc.
```

Comment:
```suggestion
- Just knowing the technology can help with identifying potentially interesting variables to explore with testing.  For example, the version of a particular technology used can help with sniffing on any present security vulnerabilities.  If you're looking to automate testing, then this can also help in deciding what tools and languages to choose for the tests. E.g., If the front-end application is written is `react-js` then writing the integration tests in `testcafe` maybe the preferred to stay in sync with the programming language and shareability of the selectors, etc.
```